dotnet/src/SemanticKernel/AI/ChatMessageGeneration/OpenAIChatHistory.cs

```diff
@@ -0,0 +1,53 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+namespace Microsoft.SemanticKernel.AI.ChatMessageGeneration;
+
+/// <summary>
+/// OpenAI Chat content
+/// See https://platform.openai.com/docs/guides/chat for details
+/// </summary>
+public class OpenAIChatHistory : ChatHistory
+{
+    private const string SystemRole = "system";
+    private const string AssistantRole = "assistant";
+    private const string UserRole = "user";
+
+    /// <summary>
+    /// Create a new and empty chat history
+    /// </summary>
+    /// <param name="assistantInstructions">Optional instructions for the assistant</param>
+    public OpenAIChatHistory(string? assistantInstructions = null)
+    {
+        if (!string.IsNullOrWhiteSpace(assistantInstructions))
+        {
+            this.AddSystemMessage(assistantInstructions);
+        }
+    }
+
+    /// <summary>
+    /// Add a system message to the chat history
+    /// </summary>
+    /// <param name="content">Message content</param>
+    public void AddSystemMessage(string content)
+    {
+        this.AddMessage(SystemRole, content);
+    }
+
+    /// <summary>
+    /// Add an assistant message to the chat history
+    /// </summary>
+    /// <param name="content">Message content</param>
+    public void AddAssistantMessage(string content)
+    {
+        this.AddMessage(AssistantRole, content);
+    }
```

Comment:
This would be better to take a username as-well,

```
public void AddAssistantMessage(string content, string username = "")
    {
        this.AddMessage(string.IsNullOrEmpty(username) ? UserRole : username, content);
    }
```

This will end users to maps users to prompts in the event of group chats.

Reply:
chatgpt doesn't yet support username so I left that out. The service supports a user id, but since it has privacy implications that should be documented, and since it's not a must have I left that out for now.